ddht/v5_1/alexandria/partials/proof.py

```diff
@@ -29,6 +36,84 @@ def __str__(self) -> str:
         return f"{display_path(self.path)}: {self.value.hex()}"
 
 
+@to_tuple
+def get_padding_elements(
+    start_index: int, num_padding_chunks: int, path_bit_length: int
+) -> Iterable[ProofElement]:
+    """
+    Get the padding elements for a proof.
+
+    By decomposing the number of chunks needed into the powers of two which
+    make up the number we can construct the minimal right hand sub-tree(s)
+    needed to pad the hash tree.
+    """
+    for power_of_two in decompose_into_powers_of_two(num_padding_chunks):
```

Comment:
I'm not sure this name is any better, but it took me a second to figure out what this method was doing. It might be more clear if this method was called `nonzero_bits`?

Reply:
Also, this is black magic! The docstring helps me understand what is happening but I think I might need a picture or a longer docstring to understand why this works.

Reply:
Yeah, I can make another pretty picture to help make it clear how/why this works.  It is indeed black magic.  Building up the intuition about how all of these different tree mechanics work was hard.  Good to write it down while I still remember it all.